gap/ReducibleMatrixGroups.gi

```diff
@@ -358,3 +358,104 @@ function(d, q, k)
 
     return MatrixGroupWithSize(F, generators, size);
 end);
+
+
+# Construction as in Proposition 4.3 of [HR05]
+BindGlobal("SpStabilizerOfIsotropicSubspace",
+function(d, q, k)
+    local field, gens, I, J, GLgen, Xi, Spgen, Yi;
+
+    if IsOddInt(d) then
+        ErrorNoReturn("<d> must even but ", d, " was given.");
+    fi;
+
+    if k >= d / 2 then
+        ErrorNoReturn("<k> must be less than <d> / 2 but <k> = ", k, 
+        " and <d> = ", d);
+    fi;
+
+    field := GF(q);
+    gens := [];
+    I := IdentityMat(d, field);
+    J := AntidiagonalMat(k, field);
+
+    # For either generator of Sp(d,q), we take an
+    # invertable matrix GLgen which acts on
+    # the first k basis vectors and puts it in
+    # a matrix with another invertable matrix acting
+    # on the last d - k basis vectors.
+    # This way, we preserve the decomposition into
+    # two isotropic subspaces.
+    # With the way we construct the second block matrix,
+    # the form is also preserved.
+    for GLgen in GeneratorsOfGroup(GL(k, q)) do
+        Xi := IdentityMat(d, field);
+        Xi{[1..k]}{[1..k]} := GLgen;
+        Xi{[d - k + 1 .. d]}{[d - k + 1 .. d]} := J * TransposedMat(GLgen ^ (-1)) * J;
+        Add(gens, Xi);
+    od;
+
+    # These two generators are diagonal block matrices with 2x2 blocks
+    # that generate a subgroup of Sp(d, q) corresponding to Sp(d - 2 * k, q).
+    for Spgen in GeneratorsOfGroup(Sp(d - 2 * k, q)) do
+        Yi := IdentityMat(d, field);
+        Yi{[k + 1 .. d - k]}{[k + 1 .. d - k]} := Spgen;
+        Add(gens, Yi);
+    od;
+
+    # This generator is mapped to matrices "with 1s down the diagonal,
+    # a (k x k) block in the middle that is symetric about the anti-diagonal,
+    # and zeros elsewhere" (cf. [HR05]) by GL(k, q) while also being fixed
+    # by Sp(d - 2k, q).
+    Add(gens, I + SquareSingleEntryMatrix(field, d, d, 1));
+
+    # This generator is in Sp(d, q) and stabilises the group generated by
+    # the first k unit vectors.
+    Add(gens, I + SquareSingleEntryMatrix(field, d, d, d - k) - SquareSingleEntryMatrix(field, d, k + 1, 1));
+
+    # Size according to Table 2.3 of [BHR13]
+    return MatrixGroupWithSize(field, gens, q ^ (k * d + QuoInt(k - 3 * k * k, 2)) * SizeGL(k, q) * SizeSp(d - 2 * k, q));
+end);
+
+
+# Construction as in Proposition 4.4 of [HR05]
+BindGlobal("SpStabilizerOfNonDegenerateSubspace",
+function(d, q, k)
+    local field, gens, twok, Spgen, Xi, Yi;
+
+    if IsOddInt(d) then
+        ErrorNoReturn("<d> must even but ", d, " was given.");
+    fi;
+
+    if k >= d / 2 then
+        ErrorNoReturn("<k> must be less than <d> / 2 but <k> = ", k, 
+        " and <d> = ", d);
+    fi;
+
+    field := GF(q);
+    gens := [];
+    twok := 2 * k;
+
+    # These generators are bidagonal block matrices
```

Comment:
```suggestion
    # These generators are diagonal block matrices
```
Or perhaps you meant something else, but I am not sure what?

Reply:
It's supposed to be block matrices with blocks on both diagonals and zeros elsewhere, but I'm not sure if "bidiagonal" is common terminology, so I'll clarify the comment.